packages/extension/src/ui/features/accountEdit/AccountEditScreen.tsx

```diff
@@ -171,3 +184,24 @@ export const AccountEditScreen: FC = () => {
     </>
   )
 }
+
+const PublicKeyCopyButton: FC<{ publicKey: string }> = ({ publicKey }) => {
+  return (
+    <CopyTooltip prompt={'Click to copy public key'} copyValue={publicKey}>
+      <Button
+        size="3xs"
+        color={"white50"}
+        bg={"transparent"}
+        _hover={{ bg: "neutrals.700", color: "text" }}
+      >
+        {`Public Key: ${shortPublicKey(publicKey)}`}
+      </Button>
+    </CopyTooltip>
+  )
+}
+
+const shortPublicKey = (publicKey: string) => {
+  const start = publicKey.slice(0, 4)
+  const end = publicKey.slice(-4)
```

Comment:
Let's use 6 characters for both public key and address. 4 might not be safe enough.